src/pool/srv_pool.c

```diff
@@ -2232,6 +2232,48 @@ ds_pool_query_handler(crt_rpc_t *rpc)
 	daos_prop_free(prop);
 }
 
+
+static int
+replace_failed_replicas(struct pool_svc *svc, struct pool_map *map)
+{
+	d_rank_list_t	*replicas;
+	d_rank_list_t	*tmp_replicas;
+	d_rank_list_t	 failed_ranks;
+	d_rank_list_t	 replace_ranks;
+	int		 rc;
+
+	rc = rdb_get_ranks(svc->ps_rsvc.s_db, &replicas);
+	if (rc != 0)
+		D_GOTO(out, rc);
+	rc = ds_pool_check_failed_replicas(map, replicas, &failed_ranks,
+					   &replace_ranks);
+	if (rc != 0) {
+		D_DEBUG(DB_MD, DF_UUID": cannot replace failed replicas: %d\n",
+			DP_UUID(svc->ps_uuid), rc);
+		D_GOTO(out, rc);
+	}
+	if (replace_ranks.rl_nr > 0)
+		ds_rsvc_add_replicas_s(&svc->ps_rsvc, &replace_ranks,
+				       ds_rsvc_get_md_cap());
+	if (failed_ranks.rl_nr > 0)
+		ds_rsvc_remove_replicas_s(&svc->ps_rsvc, &failed_ranks);
+	D_FREE(failed_ranks.rl_ranks);
+	memset(&failed_ranks, 0, sizeof(failed_ranks));
```

Comment:
why do you memset these stack variables?   I don't see them used after the memset

Reply:
You're right. It is probably not needed; maybe I was going to use it somewhere but dropped the idea.